### DISCUSSION TITLE
What conditions need to be satisfied for a Buffer entry to be evicted from the BufferPool when its running out of Memory?

### DISCUSSION THREAD

**Discussion by user8555** [duckdb/duckdb#12016] [Q&A]
I'm running into OOMs #12010 and when I look at the code, it looks like it tried to evict, but failed to do so.

**Comment by Tishj** [comment]
There need to be unpinned blocks, meaning the blocks are no longer in use.
If it failed to evict then that means all blocks are in use, or at least, are still pinned.

**Comment by user8555** [comment]
> If it failed to evict then that means all blocks are in use, or at least, are still pinned.

I'm just running INSERT OR UPDATE statements or I'm running the Appender.
There are no active SELECT statements.

All the blocks must be unpinnable right? Can you not write any block to temp file as long as there are no active readers? Or is a block pinned as long as the transaction is not committed?

This is a serious issue for me. I'm just unable to run this app for more than 2 hours in continuous ingestion mode.

The documentation suggests reducing the memory_limits. Which I did. Still of no use.

Is there any way to fix this? Or will this always OOM?

**Comment by user8555** [comment]
In my scenario, I;ve attached 1024 Databases and I'm running an Appender on each of them in parallel/concurrently. 

So, is it possible that the other 1023 transactions are open / appender is not flushed and so they are holding onto blocks that cannot be released until the transaction is committed? So, will it improve my condition if I shorten the size of those transactions such that this Appender gets some memory to free up?
